hubspot/plugin.py

```diff
@@ -770,50 +773,77 @@ def _search_contacts(
             )
             contacts.append(contact)
 
-        next_pagination_token = (
-            HubSpotPaginationToken(token=data["paging"]["next"]["after"])
-            if "paging" in data and "next" in data["paging"]
-            else None
-        )
+        token = data.get("paging", {}).get("next", {}).get("after")
+        next_pagination_token = HubSpotPaginationToken(token=token) if token else None
         return contacts, next_pagination_token
 
 
+@dataclass
+class HubSpotSearchCondition:
+    property_name: str
+    operator: Literal[
+        "EQ",
+        "NEQ",
+        "LT",
+        "LTE",
+        "GT",
+        "GTE",
+    ]
+    value: HubSpotPropertyValue
+
+
+@dataclass
+class HubSpotAndFilterGroup:
+    and_conditions: List[HubSpotSearchCondition]
+
+
 @purpose("Search contacts.")
 def hubspot_search_contacts(
-    search_criteria: Mapping[str, str],
+    or_filter_groups: List[HubSpotAndFilterGroup],
```

Comment:
does this work well? it reads quite confusingly to me, and i suspect code gen will be confused too.

i think we might be able to get pretty far with only a bunch of AND conditions without the or around the groups?